src/manifolds/Decorator.jl

```diff
@@ -0,0 +1,560 @@
+#
+# General functions for decorators
+#
+"""
+    AbstractDecoratorManifold<:Manifold
+
+"""
+abstract type AbstractDecoratorManifold<:Manifold end
+"""
+    is_default_decorator(M)
+
+For any manifold that [`is_decorator_manifold`](@ref) it might be the default
+decorator. This function dispatches onto these checks for specific decorators
+"""
+is_default_decorator(M::Manifold) = false
+
+"""
+    is_decorator_transparent(M, f)
+
+Given a [`Manifold`](@ref) `M` and a function `f`, indicate, whether a
+[`is_decorator_manifold`](@ref) acts transparent for `f`. This means, it
+just passes through down to the internally stored manifold.
+Only decorator manifolds can be transparent and their default is, to be transparent.
+A function that is affected by the decorator hence has to set this to `Val(false)`
+actively.
+If a decorator manifold is not in general transparent, it might still pass down
+for the case that a decorator is the default decorator, see [`is_default_decorator`](@ref).
+"""
+is_decorator_transparent(M::Manifold, f) = is_decorator_transparent(M::Manifold, f, is_decorator_manifold(M))
+is_decorator_transparent(M::Manifold, f, ::Val{true}) = true
+is_decorator_transparent(M::Manifold, f, ::Val{false}) = false
+
+
+function manifold_function_not_implemented_message(M,f,x...)
+    s = join(string.([x...]),", "," and ")
```

Comment:
Previously we were just printing types of arguments:
```suggestion
    s = join(map(string, map(typeof, x)),", "," and ")
```

Reply:
By the way, this function is a very good idea and should definitely be in `ManifoldsBase.jl`

Reply:
Thanks! I noticed that I was typing the same error message again and again. I will move that when this one is feature-complete, since I would even like to move all decorator-stuff to `Base`, too, such that parts of it can be used for `ArrayManifold`, though that one overwrites most anyways.

Reply:
And yes, I was planning to print types, so your suggestion is completely right.